mp/src/game/server/momentum/mom_timer.cpp

```diff
@@ -32,6 +32,26 @@ CMomentumTimer::CMomentumTimer() : CAutoGameSystemPerFrame("CMomentumTimer"),
 {
 }
 
+void CvarChangeCallback(IConVar* var, const char* pOldValue, float flOldValue)
+{
+    static bool s_bCvarReverted = false;
+
+    if (g_pMomentumTimer->IsRunning() && FStrEq(var->GetName(), "fps_max") && !s_bCvarReverted)
+    {
+        // Can't hook the change before it happens hence this jank
+        s_bCvarReverted = true;
+        var->SetValue(flOldValue);
+        s_bCvarReverted = false;
+        Warning("Cannot change fps_max while the timer is running!\n");
+    }
+}
+
+bool CMomentumTimer::Init()
+{
+    g_pCVar->InstallGlobalChangeCallback(CvarChangeCallback);
+    return true;
+}
```

Comment:
You can actually use the ConVar_Validated macro `CREATE_CONVAR_CV` to create a validated ConVar that can reject the change upfront if the value is trying to change when the timer is running.

See the [`mom_rj_center_fire`](https://github.com/momentum-mod/game/blob/develop/mp/src/game/shared/momentum/weapon/weapon_mom_rocketlauncher.cpp#L28) convar for more info on how to do it.

Reply:
Ah crap I just realized this is for an existing ConVar. I wonder if something [like this](https://gist.github.com/GamerDude27/e0404472d51d8f11ff4835f140d6a659#file-menu_background-cpp-L341) instead would be possible for a ConVar.

Reply:
But `fps_max` is defined in the engine so this wouldn't work.

Reply:
Never mind, I was able to adapt that method a bit to make it work here.